src/System.IO.Pipelines.Networking.Tls/Internal/OpenSsl/Interop.cs

```diff
@@ -0,0 +1,451 @@
+        public static void SSL_free(IntPtr ssl)
+        {
+            if (IsWindows)
+            {
+                WindowsLib.SSL_free(ssl);
+            }
+            else if (IsOsx)
+            {
+                OsxLib.SSL_free(ssl);
+            }
+            else
+            {
+                UnixLib.SSL_free(ssl);
+            }
+        }
+        public static IntPtr SSL_new(IntPtr sslContext) => IsWindows ? WindowsLib.SSL_new(sslContext) : IsOsx ? OsxLib.SSL_new(sslContext) : UnixLib.SSL_new(sslContext);
+        public static void SSL_CTX_free(IntPtr sslCtx)
+        {
+            if (IsWindows)
+            {
+                WindowsLib.SSL_CTX_free(sslCtx);
+            }
+            else if (IsOsx)
+            {
+                OsxLib.SSL_CTX_free(sslCtx);
+            }
+            else
+            {
+                UnixLib.SSL_CTX_free(sslCtx);
+            }
+        }
+        public static void SSL_get0_alpn_selected(IntPtr ssl, out byte* data, out int len)
+        {
+            if (IsWindows)
+            {
+                WindowsLib.SSL_get0_alpn_selected(ssl, out data, out len);
+            }
+            else if (IsOsx)
+            {
+                OsxLib.SSL_get0_alpn_selected(ssl, out data, out len);
+            }
+            else
+            {
+                UnixLib.SSL_get0_alpn_selected(ssl, out data, out len);
+            }
+        }
+        public static void SSL_CTX_set_alpn_select_cb(OpenSslSecurityContext context)
+        {
+            if (IsWindows)
+            {
+                WindowsLib.SSL_CTX_set_alpn_select_cb(context._sslContext, context._alpnCallback, IntPtr.Zero);
+            }
+            else if (IsOsx)
+            {
+                OsxLib.SSL_CTX_set_alpn_select_cb(context._sslContext, context._alpnCallback, IntPtr.Zero);
+            }
+            else
+            {
+                UnixLib.SSL_CTX_set_alpn_select_cb(context._sslContext, context._alpnCallback, IntPtr.Zero);
+            }
+        }
+        public static int SSL_CTX_set_alpn_protos(IntPtr ctx, IntPtr protocolList, uint protocolListLength) => IsWindows ? WindowsLib.SSL_CTX_set_alpn_protos(ctx, protocolList, protocolListLength) : IsOsx ? OsxLib.SSL_CTX_set_alpn_protos(ctx, protocolList, protocolListLength) : UnixLib.SSL_CTX_set_alpn_protos(ctx, protocolList, protocolListLength);
+        public static CipherInfo GetCipherInfo(IntPtr ssl) => IsWindows ? WindowsLib.GetCipherInfo(ssl) : IsOsx ? OsxLib.GetCipherInfo(ssl) : UnixLib.GetCipherInfo(ssl);
+
+        const int SSL_CTRL_OPTIONS = 32;
+        const ContextOptions Default_Context_Options = ContextOptions.SSL_OP_NO_SSLv2 | ContextOptions.SSL_OP_NO_SSLv3;
+
+        [Flags]
+        public enum ContextOptions : long
```

Comment:
Not sure what u mean.. could be missing something as I am on a phone on the tube.. being able to tweak options needs to added as does session resumption and other stuff. As far as I can tell I disable ssl2  and ssl3 cause they are broken and shouldn't be touched... pit of success and all